### ISSUE TITLE
Rule proposal: prefer placing media query in rules

### ISSUE THREAD

**Issue by golopot** [stylelint/stylelint#4188]
> What is the problem you're trying to solve?

In sass, media queries can be placed inside of rules, some authors prefers doing so instead of the opposite, because it keep codes for a rule in one place, in other words, avoid scattering related codes.

### Pass
Declarations for the class `.aaa` are concentrated in on place.

```scss
.aaa {
  ...
  @media (min-width: 900px) {
    ...
  }
  @media (min-width: 400px) {
    ...
  }
}

.bbb {
  ...
  @media (min-width: 900px) {
    ...
  }
  @media (min-width: 400px) {
    ...
  }
}
```
### Fail
Declarations for the class `.aaa` are scattered.

```scss
.aaa {
  ... // scattered here.
}

.bbb {
  ...
}

@media (min-width: 400px) {
  .aaa {
    ... // scattered here.
  }

  .bbb {
    ...
  }
}

@media (min-width: 900px) {
  .aaa {
    ... // scattered here.
  }

  .bbb {
    ...
  }
}
```
> What solution would you like to see?

A new rule to forbid media queries containing rules.

```scss
@media (min-width: 400px) {
// ~~~~~~~~~~~~~~~~~~~~~~ Error: It is preferred to place media query inside of rules.
  .aaa {
    ...
  }
  .bbb {
    ...
  }
}
```

**Comment by hudochenkov** [comment]
I think this rule should be a plugin or part of [stylelint-scss](https://github.com/kristerkari/stylelint-scss), because I can't come up with a use case for standard CSS.

**Comment by jeddy3** [comment]
> I think this rule should be a plugin

Agreed.

I think the [stylelint-use-nesting plugin](https://github.com/csstools/stylelint-use-nesting) supports this nested at-rule use case (because there was talk of supporting nested at-rules in the spec, although [this might change at some point](https://github.com/jonathantneal/postcss-nesting#postcss-nesting-). 

Either way I'll close the issue as it's best handled in a plugin, like stylelint-use-nesting or something similar.
